File: perl/4_sum_of_primes.pl
#!/usr/bin/env perl

use strict;
use warnings;

sub is_prime {
	my $n = shift @_;

	return 0 if $n <= 1;
	return 1 if $n == 2;
	return 0 if $n % 2 == 0;

	my $sqrt = sqrt $n;
	for ( my $i = 3; $i <= $sqrt; $i += 2 ) {
		return 0 if $n % $i == 0;
	}

	return 1;
}

my $sum   = 2;
my $count = 1;
my $i     = 1;
while ( $count != 1000 ) {
	$i += 2;
	next if not is_prime( $i );

	$sum += $i;
	$count++;
}

print "$sum\n";

File: python/4_sum_of_primes.py
#!/usr/bin/env python

import math

def is_prime(n):
    if n <= 1:
        return False
    elif n == 2:
        return True
    elif n % 2 == 0:
        return True

    sqrt = math.sqrt(n)
    i    = 3
    while i <= sqrt:
        if n % i == 0:
            i += 2
            return False
        i += 2

    return True

sum   = 2
count = 1
i     = 1
while count != 1000:
    i += 2

    if not is_prime(i):
        continue

    count += 1
    sum   += i

print sum

